# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteVideoDecoder.cpp
# Commit: afef5857aea6
# Full Hash: afef5857aea6cbbd3e9296ce39c004812967d095
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2022-09-08 04:52:24
# Regressor Bug: 1787561
# File Overlap Count: 1
# Description:
#   Bug 1787561 Part 1: Make GPUVideoImage track color depth, and make RemoteImageHolder pass it through. r=media-playback-reviewers,gfx-reviewers,alwu,aosmond
#   
#   This ensures that the Image that the client receives has the color depth
#   information that was sent to the GPU, even though the client can't query
#   the GPU directly.
# ==============================================================================

diff -r 1b2dc91d3c38 -r afef5857aea6 dom/media/ipc/RemoteVideoDecoder.cpp
--- a/dom/media/ipc/RemoteVideoDecoder.cpp	Wed Sep 07 18:23:24 2022 +0000
+++ b/dom/media/ipc/RemoteVideoDecoder.cpp	Wed Sep 07 18:31:45 2022 +0000
@@ -274,14 +274,10 @@
         MediaDataIPDL(data->mOffset, data->mTime, data->mTimecode,
                       data->mDuration, data->mKeyframe),
         video->mDisplay,
-        RemoteImageHolder(
-            mParent,
-            XRE_IsGPUProcess()
-                ? VideoBridgeSource::GpuProcess
-                : (XRE_IsRDDProcess()
-                       ? VideoBridgeSource::RddProcess
-                       : VideoBridgeSource::MFMediaEngineCDMProcess),
-            size, sd),
+        RemoteImageHolder(mParent,
+                          XRE_IsGPUProcess() ? VideoBridgeSource::GpuProcess
+                                             : VideoBridgeSource::RddProcess,
+                          size, video->mImage->GetColorDepth(), sd),
         video->mFrameID);
 
     array.AppendElement(std::move(output));